lib/temporal/workflow/errors.rb

```diff
@@ -67,12 +67,19 @@ def self.generate_error_for_child_workflow_start(cause, workflow_id)
         end
       end
 
+      def self.generate_error_for_child_workflow_timeout
+        ChildWorkflowTimeoutError.new('The child workflow timed out before succeeding')
+      end
+
+      def self.generate_error_for_child_workflow_terminated
+        ChildWorkflowTerminatedError.new('The child workflow was terminated')
+      end
```

Comment:
nit: these 2 methods provide very little value that might not be worth the added indirection

Reply:
dont disagree but just followed the same pattern to what we have today! What do you think?

Reply:
I don't think there's much benefit in doing this for every single error type, so we can probably break the patter for these simpler use-cases

Reply:
Let me update! :)